File: domain/src/main/java/com/github/dkurata38/liblio/domain/user/UserRepository.java
package com.github.dkurata38.liblio.domain.user;

import java.util.Optional;

public interface UserRepository {
    /**
     * @param username not {@code null}
     * @return user
     */
    Optional<User> findByUsername(Username username);

    /**
     * @param user not {@code null}
     */
    void create(User user);
}

File: domain/src/main/kotlin/com/github/dkurata38/liblio/domain/user/UserRepository.kt
package com.github.dkurata38.liblio.domain.user

interface UserRepository {
    /**
     * @param username not `null`
     * @return user
     */
    fun findByUsername(username: Username): User?

    /**
     * @param user not `null`
     */
    fun create(user: User)
}
